pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -541,6 +569,38 @@ func removeFinalizerFromCRDInstance(ctx context.Context,
 	}
 }
 
+// addFinalizerToPVC will add the CNS Finalizer, cns.vmware.com/pvc-protection,
+// from a given PersistentVolumeClaim.
+func addFinalizerToPVC(ctx context.Context, client client.Client, pvc *v1.PersistentVolumeClaim) {
+	log := logger.GetLogger(ctx)
+	pvc.Finalizers = append(pvc.Finalizers, cnsoperatortypes.CNSPvcFinalizer)
+	log.Debugf("Removing %q finalizer from PersistentVolumeClaim: %q on namespace: %q",
+		cnsoperatortypes.CNSPvcFinalizer, pvc.Name, pvc.Namespace)
+	err := client.Update(ctx, pvc)
+	if err != nil {
+		log.Errorf("failed to update PersistentVolumeClaim: %q on namespace: %q. Error: %+v",
+			pvc.Name, pvc.Namespace, err)
+	}
+}
+
+// removeFinalizerFromPVC will remove the CNS Finalizer, cns.vmware.com/pvc-protection,
+// from a given PersistentVolumeClaim.
+func removeFinalizerFromPVC(ctx context.Context, client client.Client, pvc *v1.PersistentVolumeClaim) {
+	log := logger.GetLogger(ctx)
+	for i, finalizer := range pvc.Finalizers {
+		if finalizer == cnsoperatortypes.CNSPvcFinalizer {
+			log.Debugf("Removing %q finalizer from PersistentVolumeClaim: %q on namespace: %q",
+				pvc.Name, pvc.Namespace)
+			pvc.Finalizers = append(pvc.Finalizers[:i], pvc.Finalizers[i+1:]...)
+		}
+	}
+	err := client.Update(ctx, pvc)
+	if err != nil {
```

Comment:
Same comment. Why are we ignoring the error here? We should fail the reconciliation so that in the next invocation we attempt to remove the finalizer.

Reply:
Yes, updated the error handling path